meta-lmp-base/recipes-containers/skopeo/skopeo/0003-increase-tcp-dialer-timeouts.patch

```diff
@@ -0,0 +1,41 @@
+From de6a62d5369bad2c32cdf8735b5195c857a8345f Mon Sep 17 00:00:00 2001
+From: Mike Sul <mike.sul@foundries.io>
+Date: Thu, 15 Jun 2023 17:48:41 +0200
+Subject: [PATCH] Increase TCP dialer timeouts
+
+Increase the TCP dialer timeouts that allegedly is supposed to minimize
+occurrence of TCP connection related errors/timeouts, including DNS
+resolution.
+
+Upstream-Status: Inappropriate [lmp specific]
+
+Signed-off-by: Mike <mike.sul@foundries.io>
+---
+ pkg/tlsclientconfig/tlsclientconfig.go | 6 +++---
+ 1 file changed, 3 insertions(+), 3 deletions(-)
+
+diff --git a/vendor/github.com/containers/image/v5/pkg/tlsclientconfig/tlsclientconfig.go b/vendor/github.com/containers/image/v5/pkg/tlsclientconfig/tlsclientconfig.go
+index 285203ba..ccea2c70 100644
+--- a/vendor/github.com/containers/image/v5/pkg/tlsclientconfig/tlsclientconfig.go
++++ b/vendor/github.com/containers/image/v5/pkg/tlsclientconfig/tlsclientconfig.go
+@@ -91,14 +91,14 @@ func hasFile(files []os.DirEntry, name string) bool {
+ // NewTransport Creates a default transport
+ func NewTransport() *http.Transport {
+ 	direct := &net.Dialer{
+-		Timeout:   30 * time.Second,
+-		KeepAlive: 30 * time.Second,
++		Timeout:   60 * time.Second,
++		KeepAlive: 60 * time.Second,
```

Comment:
Seems sane. Might be useful to cross-reference with what we use for fioconfig?

Reply:
We don't set the dialer timeout in fioconfig. The latter sets only `http.Client.Timeout` which is not appropriate for skopeo since it includes a response delivery time and skopeo has to wait until a blob is pulled...

Reply:
Looks like the default [dialer](https://go.dev/src/net/http/transport.go) uses:
```
var DefaultTransport RoundTripper = &Transport{
	Proxy: ProxyFromEnvironment,
	DialContext: defaultTransportDialContext(&net.Dialer{
		Timeout:   30 * time.Second,
		KeepAlive: 30 * time.Second,
	}),
	ForceAttemptHTTP2:     true,
	MaxIdleConns:          100,
	IdleConnTimeout:       90 * time.Second,
	TLSHandshakeTimeout:   10 * time.Second,
	ExpectContinueTimeout: 1 * time.Second,
}
```

so - should we change any of these fields? and - orthogonal - but should apply similar changes to fioconfig?

Reply:
This PR changes the dialer's default `30` to `60. The rest fields are not related to the dialer.

> but should apply similar changes to fioconfig?

I don't know, `fioconfig` keeps trying too, so maybe just nice to have.

Reply:
@doanac Do you think it makes sense to change the default dialer's timeouts to 60s in fioconfig too?

Reply:
if it helps, let's do it.